src/Symfony/Component/HttpKernel/Kernel.php

```diff
@@ -469,6 +463,32 @@ protected function initializeBundles()
     }
 
     /**
+     * @param BundleInterface $bundle
+     * @param bool $dependency
+     */
+    protected function addBundle($bundle, $dependency = false)
+    {
+        $name = $bundle->getName();
+
+        if (isset($this->bundles[$name]) && false === $dependency) {
```

Comment:
Using two bundles with the same name does not only impact this feature, but what about templating ? When using `AppBundle:Default:index.html.twig` can be a real problem if you have these two FQCN:
\AppBundle
\VendorName\CustomerApplication\Bundle\AppBundle
 
These FQCN are entirely different, but the bundle name remains the same, so this is still a problem. I only set up this check the same way it was before. Just look at [this diff](https://github.com/symfony/symfony/pull/13506/files#diff-50575470114d078b3ca3622ae30fba6eL427) :)